### PR TITLE
added Page Factory pattern Test

### PR DESCRIPTION
**Pull request by OlgaAsadchaya**

### CODE REVIEW

File: src/com/lesson/TutByPage2.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public class TutByPage2 {
```
Comment (EugenBorisik) [inline src/com/lesson/TutByPage2.java:13]: Rename

File: src/com/lesson/TutByPage2.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    private WebElement password;
+
+    @FindBy(css = "input.button.auth__enter")
+    private WebElement enterButton;
+
+    @FindBy(className = "uname")
+    private WebElement content;
+
+    private WebDriver driver;
```
Comment (EugenBorisik) [inline src/com/lesson/TutByPage2.java:28]: I recommend to make WebDriver instance final in page objects, so you will never change it

File: src/com/lesson/TutByPage2.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    private String url = "https://tut.by";
```
Comment (EugenBorisik) [inline src/com/lesson/TutByPage2.java:29]: Make constant

File: src/com/lesson/TutByPage2.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+
+    public TutByPage2() {
+        driver = new ChromeDriver();
```
Comment (EugenBorisik) [inline src/com/lesson/TutByPage2.java:33]: Move driver initialization in separate class (create singleton for it).

File: src/com/lesson/TutByPage2.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        PageFactory.initElements(driver, this);
+    }
+
+    public void load() {
+        this.driver.get(url);
+    }
+    public void close() {
+        this.driver.close();
```
Comment (EugenBorisik) [inline src/com/lesson/TutByPage2.java:41]: This method will be moved from page object into your singleton for WebDriver. You must understand that Page object encapsulate actions on the pages - here you manipulate with browser, so it is out of scope of page object class

File: src/com/lesson/TutByPage2.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    public void login() {
+        enterPopup.click();
+        login.sendKeys("seleniumtests@tut.by");
```
Comment (EugenBorisik) [inline src/com/lesson/TutByPage2.java:46]: Lines 46-47: you use hard-coded parameters and there is no way to login with another params. Move this logic into parameters for method ... login(String user, String password)

File: src/com/lesson/TutByPage2.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    private String url = "https://tut.by";
+
+
+    public TutByPage2() {
+        driver = new ChromeDriver();
+        PageFactory.initElements(driver, this);
+    }
+
+    public void load() {
+        this.driver.get(url);
```
Comment (EugenBorisik) [inline src/com/lesson/TutByPage2.java:38]: Logic with opening start page can be moved in to constructor in this case

**Review by EugenBorisik — CHANGES_REQUESTED** [review]
Please update code according to comments.
Let's imagine the following: when you are on the page https://tut.by - it is TytByPage. After your perform login (as far as I remember only header of the page changes) - it is HomePage. I want such changes for training purposes (in real project it will look in similar way) + it will help you to better understand Page Object pattern. 
So, final state:
1. Singleton for WebDriver
2. 2 page object classes: TytByPage and HomePage
3. Test class with 2 tests: login and logout.

**Comment by OlgaAsadchaya** [comment]
Closed this pull request because have created new pull request with new brunch (I got confused )

**Pull request closed without merging**